src/python/pants/core/register.py

```diff
@@ -51,3 +54,25 @@ def rules():
 
 def target_types():
     return [ArchiveTarget, Files, GenericTarget, Resources, RelocatedFiles]
+
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class CoreSessionLifecycleHandler(SessionLifecycleHandler):
+    def on_session_start(self):
+        logger.info("on_session_start")
+
+    def on_session_end(self):
+        logger.info("on_session_end")
+
+
+class CoreLifecycleHandler(ExtensionLifecycleHandler):
+    def on_session_create(self, options: Options) -> Optional[SessionLifecycleHandler]:
+        logger.info("on_session_create")
+        return CoreSessionLifecycleHandler()
+
+
+def lifecycle_handlers():
+    return [CoreLifecycleHandler()]
```

Comment:
This change is just to provide a quick example of how a lifecycle handler would work in practice. It would be removed in the final version of the PR.

Reply:
Removed this.